File: app/src/main/java/com/ed/redditapp/AppDaggerComponent.java
package com.ed.redditapp;

import com.ed.redditapp.lib.api.RedditApi;
import com.ed.redditapp.lib.api.RedditApiDaggerModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {RedditApiDaggerModule.class})
public interface AppDaggerComponent {
    RedditApi redditApi();
}

File: app/src/main/java/com/ed/redditapp/AppDaggerComponent.kt
package com.ed.redditapp

import com.ed.redditapp.lib.api.RedditApi
import com.ed.redditapp.lib.api.RedditApiDaggerModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [RedditApiDaggerModule::class])
interface AppDaggerComponent {
    fun redditApi(): RedditApi
}